scripts/exglobal_prep_sfc.sh

```diff
@@ -0,0 +1,196 @@
+export MODEL_SNOW_FILE=${RUN}.${cycle}.snogrb_t${CASE:1}.${LONB_CASE}.${LATB_CASE}
+export MODEL_SNOW_FILE_PREV=${COMINobsproc_PREV}/${RUN}.${gcycle}.snogrb_t${CASE:1}.${LONB_CASE}.${LATB_CASE}
+
+echo "Create ${CASE} snow data."
+"${USHgfs}/prep_sfc_snow.sh"
+export err=$?
+
+#----------------------------------------------------------------------
+# If there was a failure in the prep_sfc_snow script, copy the 6-hr old
+# snow file to the current file.  The gfs/gdas can run with old snow data
+# for a day or two at most.   So while not fatal, any errors must be
+# investigated
+#
+# If there is a failure in the prep_sfc_snow script AND the 6-hour old
+# snow file is not available as a backup, abort the script.  The
+# global cycling can't run without an snow analysis.
+#-----------------------------------------------------------------------
+
+if [[ ${err} -ne 0 ]]; then
+  if [[ -s "${MODEL_SNOW_FILE_PREV}" ]]; then
+    echo "COPY OLD ${CASE} SNOW FILE TO CURRENT DIRECTORY"
+    cpfs "${MODEL_SNOW_FILE_PREV}" "${COMOUT_OBS}/${MODEL_SNOW_FILE}"
+  else
+    err_exit "CURRENT AND 6-HR OLD ${CASE} SNOW MISSING, ABORT!"
+  fi  # check of missing 6-hr snow file
+fi  # return code check
+
+#-----------------------------------------------------------------------
+# The GFS step does not run the enkf stuff.  So there is no need to
+# create an enkf snow file.
+#-----------------------------------------------------------------------
+
+if test "${RUN}" = "gfs"
+then
+  exit 0
+fi
+
+LONB_CASE_ENS=$((4*${CASE_ENS:1}))
+LATB_CASE_ENS=$((2*${CASE_ENS:1}))
+
+export MODEL_SLMASK_FILE=${SLMASK_ENKF:-${FIXgfs}/am/global_slmask.t${CASE_ENS:1}.${LONB_CASE_ENS}.${LATB_CASE_ENS}.grb}
+export MODEL_LATITUDE_FILE=${MDL_LATS_ENKF:-${FIXgfs}/am/global_latitudes.t${CASE_ENS:1}.${LONB_CASE_ENS}.${LATB_CASE_ENS}.grb}
+export MODEL_LONGITUDE_FILE=${MDL_LONS_ENKF:-${FIXgfs}/am/global_longitudes.t${CASE_ENS:1}.${LONB_CASE_ENS}.${LATB_CASE_ENS}.grb}
+export GFS_LONSPERLAT_FILE=${LONSPERLAT_ENKF:-${FIXgfs}/am/global_lonsperlat.t${CASE_ENS:1}.${LONB_CASE_ENS}.${LATB_CASE_ENS}.txt}
+export MODEL_SNOW_FILE=${RUN}.${cycle}.snogrb_t${CASE_ENS:1}.${LONB_CASE_ENS}.${LATB_CASE_ENS}
+export MODEL_SNOW_FILE_PREV=${COMINobsproc_PREV}/${RUN}.${gcycle}.snogrb_t${CASE_ENS:1}.${LONB_CASE_ENS}.${LATB_CASE_ENS}
+
+echo "Create enkf snow data."
+"${USHgfs}/prep_sfc_snow.sh"
+export err=$?
+
+#-----------------------------------------------------------------------
+# Check for errors creating enkf snow.  Use 6-hour old data
+# as backup.  If old data not available, abort.
+#-----------------------------------------------------------------------
+
+if [[ ${err} -ne 0 ]]; then
+  if [[ -s "${MODEL_SNOW_FILE_PREV}" ]]; then
+    echo "COPY OLD ENKF SNOW FILE TO CURRENT DIRECTORY"
+    cpfs "${MODEL_SNOW_FILE_PREV}" "${COMOUT_OBS}/${MODEL_SNOW_FILE}"
+  else
+    err_chk "FATAL ERROR: CURRENT AND 6-HR OLD ENKF SNOW MISSING"
```

Comment:
No need to add `FATAL ERROR`.  `err_chk` and `err_exit` prepend that.  Also, prefer `err_exit` over `err_chk`

```suggestion
    err_exit "CURRENT AND 6-HR OLD ENKF SNOW MISSING"
```

Reply:
Committed @ e88b3022b.